### PR TITLE
Initial check state

### PR DESCRIPTION
**Pull request by hoffoo**
This is an whack at #1239

Thanks!

### CODE REVIEW

File: jobspec/parse.go [hunk 1]
```diff
@@ -770,6 +770,7 @@ func parseChecks(service *structs.Service, checkObjs *ast.ObjectList) error {
 			"port",
 			"command",
 			"args",
+			"status",
```
Comment (diptanu) [inline jobspec/parse.go]: Can you add this to the parse test suite?

File: nomad/structs/structs.go [hunk 2]
```diff
@@ -1653,6 +1654,15 @@ func (sc *ServiceCheck) validate() error {
 		return fmt.Errorf("interval (%v) can not be lower than %v", sc.Interval, minCheckInterval)
 	}
 
+	switch sc.Status {
+	case "":
+	case "passing":
```
Comment (diptanu) [inline nomad/structs/structs.go]: Use the constants defined here - https://godoc.org/github.com/hashicorp/consul/api#pkg-constants

**Comment by diptanu** [comment]
@hoffoo Thanks! Added some comments, we are going to merge this after we get 0.4.1 out of the door.

File: jobspec/parse.go (hunk 1 above)
Comment (dadgar) [inline jobspec/parse.go]: I also think it is a little vague of a name. Maybe `initial_status`

Comment (hoffoo) [inline jobspec/parse.go]: @dadgar should I rename it to InitialStatus everywhere? or just in parse.go?

Comment (dadgar) [inline jobspec/parse.go]: Everywhere for consistency please

**Comment by hoffoo** [comment]
okay i made the changes, please let me know of anything else

Thanks!

**Comment by dadgar** [comment]
@hoffoo Could you rebase and get this mergable?

**Comment by hoffoo** [comment]
done! do you want me to open a squashed pr? not sure what the process is

**Comment by diptanu** [comment]
@hoffoo So your branch is behind our master, which is why there are so many extra commits in your PR which have already landed in `master`. I would suggest rebasing your master with our master, and then rebasing your branch `initial_check_state` with your master(once that is up to date with the commits from our `master`). We should be able to get your PR in fairly quickly after that.

**Comment by hoffoo** [comment]
@diptanu sorry about that, new commit is ready

**Pull request closed without merging**